tests/apiclient/test_vcs.py

```diff
@@ -24,27 +24,52 @@
 """
 Tests for VCS utility classes and functions.
 """
-
+import base64
 from textwrap import dedent
 import subprocess
 import unittest
-from unittest.mock import patch, PropertyMock
+from unittest.mock import Mock, patch, PropertyMock
 
 from sat.apiclient.vcs import VCSError, VCSRepo
 
 
 class TestVCSRepo(unittest.TestCase):
     """Tests for the VCSRepo class."""
     def setUp(self):
+        self.repo_name = 'sat-config-management'
+        self.repo_path = f'/vcs/cray/{self.repo_name}.git'
+        self.vcs_host = 'api-gw-service-nmn.local'
+        self.clone_url = f'https://{self.vcs_host}{self.repo_path}'
+        self.vcs_repo = VCSRepo(self.clone_url)
+
+        self.vcs_username = 'crayvcs'
+
         self.mock_subprocess_run = patch('sat.apiclient.vcs.subprocess.run').start()
+        self.mock_load_kube_config = patch('sat.apiclient.vcs.load_kube_config').start()
+        self.mock_k8s_api_cls = patch('sat.apiclient.vcs.CoreV1Api').start()
+        self.mock_k8s_api = self.mock_k8s_api_cls.return_value
+        mock_vcs_secret_data = {'vcs_username': base64.b64encode(self.vcs_username.encode('utf-8'))}
+        self.mock_k8s_api.read_namespaced_secret.return_value = Mock(data=mock_vcs_secret_data)
 
     def tearDown(self):
         patch.stopall()
 
-    def test_default_username(self):
-        """Test that the VCSRepo constructor sets the default username when None supplied"""
-        repo = VCSRepo('foo/bar.git')
-        self.assertEqual(repo.username, VCSRepo._default_username)
+    def test_repo_path(self):
+        """Test repo_path property of VCSRepo."""
+        self.assertEqual(self.repo_path, self.vcs_repo.repo_path)
+
+    def test_repo_name(self):
+        """Test repo_name property of VCSRepo."""
+        self.assertEqual(self.repo_name, self.vcs_repo.repo_name)
+
+    def test_vcs_username(self):
```

Comment:
w.r.t. the username-in-URL bug above, there should probably be a unit test to check that behavior is correct.

Reply:
Done, thanks!